model/src/cfd.rs

```diff
@@ -796,11 +809,24 @@ impl Cfd {
         Ok(())
     }
 
-    fn can_settle_collaboratively(&self) -> bool {
-        !self.is_closed()
-            && !self.commit_finality
-            && !self.is_attested()
-            && !self.is_in_force_close()
+    fn can_settle_collaboratively(&self) -> Result<(), CannotSettleCollaborativeley> {
```

Comment:
🙃 Since it doesn't return a boolean anymore should we call it `check_collaborative_settlement` or something?

Reply:
I'd keep it for now, given that we also have `can_rollover` with the same pattern, and not other `can_xxx` pattern for similar logic.